### PR TITLE
Enforce TLS 1.2 for libcurl and winHttp

### PR DESCRIPTION
**Pull request by vhvb1989**
Enforce libcurl and winHttp transports to use TLS version 1.2 when creating a new https request.

### CODE REVIEW

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -3,10 +3,13 @@
 ## 1.5.0-beta.1 (Unreleased)
 
 ### Features Added
+
 - When a `RequestFailedException` exception is thrown, the `what()` method now includes information about the HTTP request which failed.
 
 ### Other Changes
```
Comment (LarryOsterman) [inline sdk/core/azure-core/CHANGELOG.md:10]: I believe that this counts as a breaking change and should be called out as such.

File: sdk/core/azure-core/src/http/curl/curl.cpp [hunk 2]
```diff
@@ -1246,8 +1246,8 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::ExtractOrCreateCurlCo
     bool resetPool)
 {
   uint16_t port = request.GetUrl().GetPort();
-  std::string const& host = request.GetUrl().GetScheme() + request.GetUrl().GetHost()
```
Comment (LarryOsterman) [inline sdk/core/azure-core/src/http/curl/curl.cpp:1249]: How did this code ever work given that GetScheme() didn't append `://` to the scheme?

Comment (LarryOsterman) [reply sdk/core/azure-core/src/http/curl/curl.cpp:1249]: Answering my own question: "host" was only ever used in output strings, except it is also used for the GetConnectionKey as the base for the key, and the key is only used when managing the connection pool. 

So the actual value of "host" is irrelevant, and thus adding the :// to the string doesn't matter.

IMHO, This is one of those cases where a brief comment ("Generate a display name for the host being connected to") or a somewhat more descriptive name (`hostDisplayName`) might make more sense - I had to spend 10 minutes reading the code trying to understand why this change was necesssary - my original thought was that the pull request changed how `GetScheme()` worked, and thus I tried to find out why this wasn't also a breaking change to the `GetScheme()` API.

**Review by LarryOsterman — APPROVED** [review]

File: sdk/core/azure-core/src/http/curl/curl.cpp (hunk 2 above)
Comment (vhvb1989) [reply sdk/core/azure-core/src/http/curl/curl.cpp:1249]: I'll move this to another PR that improves a Transport Exception output

Comment (vhvb1989) [reply sdk/core/azure-core/src/http/curl/curl.cpp:1249]: https://github.com/Azure/azure-sdk-for-cpp/pull/3456

**Comment by vhvb1989** [comment]
https://github.com/Azure/azure-sdk-for-cpp/pull/3456 should go before this

**Pull request closed without merging**